src/trackinfoobject.cpp

```diff
@@ -566,46 +562,65 @@ void TrackInfoObject::setTrackNumber(const QString& s) {
     }
 }
 
-int TrackInfoObject::getTimesPlayed() const {
+void TrackInfoObject::setPlayed(bool played) {
     QMutexLocker lock(&m_qMutex);
-    return m_iTimesPlayed;
+    PlayCounter playCounter(m_playCounter);
+    playCounter.setPlayed(played);
+    if (m_playCounter != playCounter) {
+        m_playCounter = playCounter;
+        setDirty(true);
+    }
 }
 
-void TrackInfoObject::setTimesPlayed(int t) {
+bool TrackInfoObject::isPlayed() const {
     QMutexLocker lock(&m_qMutex);
-    if (t != m_iTimesPlayed) {
-        m_iTimesPlayed = t;
+    return m_playCounter.isPlayed();
+}
+
+void TrackInfoObject::setTimesPlayed(int timesPlayed) {
+    QMutexLocker lock(&m_qMutex);
+    PlayCounter playCounter(m_playCounter);
+    playCounter.setTimesPlayed(timesPlayed);
+    if (m_playCounter != playCounter) {
+        m_playCounter = playCounter;
         setDirty(true);
     }
 }
 
-void TrackInfoObject::incTimesPlayed() {
-    setPlayedAndUpdatePlaycount(true);
+void TrackInfoObject::resetTimesPlayed() {
+    QMutexLocker lock(&m_qMutex);
+    PlayCounter playCounter(m_playCounter);
+    playCounter.resetTimesPlayed();
```

Comment:
Maybe we should get rid of resetTimesPlayed() and here. It is confusing since the played counter is not relevant for "dirty".